SimTKcommon/src/Pathname.cpp

```diff
@@ -238,6 +240,161 @@ void Pathname::deconstructPathname( const string&   name,
+        bool dontApplySearchPath;
+        deconstructPathname(path, dontApplySearchPath, directory, fileName, extension);
+        if (!dontApplySearchPath)
+            directory = getCurrentWorkingDirectory() + directory;
+
+    // swd was not empty. If necessary, preprocess the swd. Then concatenate swd and path.
+    // Finally, resolve drives.
+    }
+    else {
+        /* PREPROCESSING THE SWD. */
+        // Preprocess the swd if it leads with a "./"
+        if (swdcleaned.substr(0, 2) == "./") {
+            swdcleaned.replace(0, 2, getCurrentWorkingDirectory(swddrive));
+            removeDriveInPlace(swdcleaned, swddrive);
+            swdcleaned.insert(0, "/"); // ensure swd starts with "/" to denote full path
+        }
+        // Also preprocess the swd if it leads with "/". Just grab current drive letter.
+        else if (swdcleaned.substr(0, 1) == "/" && swddrive.empty()) {
+            swddrive = getCurrentDriveLetter();
+        }
+        // Also preprocess if swd starts with something of the form "C:folder/". Resolve
+        // by finding the current directory of this drive.
+        else if (!swddrive.empty()) {
+            swdcleaned.insert(0, getCurrentWorkingDirectory(swddrive));
+            removeDriveInPlace(swdcleaned, swddrive);
+            swdcleaned.insert(0, "/"); // ensure swd starts with "/" to denote full path
+        }
+
+        /* CHECKING IF THE SWD SHOULD BE PREPENDED TO THE PATH. */
+        // If path was an absolute path (on Windows this includes a drive or begins with "@/")
+        // then we should not prepend the swd.
+        if (pathcleaned.substr(0, 1) == "/" || pathcleaned.substr(0, 2) == "@/") {
+            processed = pathcleaned;
+            // Ensure that if there was a pathdrive, ignore the swddrive.
+            if (!pathdrive.empty()) swddrive = "";
+        }
+
+        // If path starts with "./", we remove the "./" and concatenate with swdcleaned.
+        else if (pathcleaned.substr(0, 2) == "./") {
+            pathcleaned.erase(0, 2);
+            processed = swdcleaned + pathcleaned;
+        }
+        // Looks like a relative path name (i.e. pathcleaned starts immediately with
+        // a directory or file name).
+        else {
+            processed = swdcleaned + pathcleaned;
+        }
+
+        // If the pathname in its entirety is just one of these, append 
+        // a slash to avoid special cases below.
+        if (processed == "." || processed == ".." || processed == "@")
+            processed += "/";
+
+        // If the path begins with "../" we'll make it ./../ to simplify handling.
+        if (processed.substr(0, 3) == "../")
+            processed.insert(0, "./");
+
+        /* RESOLVING THE FULL PATH */
+        // Full path is determined except for drive letter. Resolve drive letter with
+        // swd, then path, then current drive.
+        if (processed.substr(0, 1) == "/") {
+            processed.erase(0, 1);
```

Comment:
You are removing a leading slash here. Speculation -- this only works because you made it "//" above per my earlier comment! I think you want to get rid of both changes. If you look at the next case below (for "@/") you can see that one leaves the "/" in place (because getThisExecutableDirectory() starts with a "/" after you remove the drive).